private/bufpkg/bufmodule/bufmoduleref/bufmoduleref.go

```diff
@@ -54,18 +54,12 @@ type FileInfo interface {
 	ExternalPath() string
 	// IsImport returns true if this file is an import.
 	IsImport() bool
-	// ModuleIdentity is the module that this file came from.
+	// ModuleIdentityOptionalCommit is the module that this file came from, along with an optional commit.
 	//
 	// Note this *can* be nil if we did not build from a named module.
 	// All code must assume this can be nil.
 	// Note that nil checking should work since the backing type is always a pointer.
-	ModuleIdentity() ModuleIdentity
-	// Commit is the commit for the module that this file came from.
-	//
-	// This will only be set if ModuleIdentity is set. but may not be set
-	// even if ModuleIdentity is set, that is commit is optional information
-	// even if we know what module this file came from.
-	Commit() string
```

Comment:
These two together sounds more like an optional `ModuleReference`, which internally already has an optional reference which might be a commit.